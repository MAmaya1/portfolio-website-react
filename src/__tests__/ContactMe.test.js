import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Import Component
import ContactMe from '../components/ContactMePage';

afterEach(cleanup);

describe('ContactMe component', () => {
    it('should render without crashing', () => {
        render(<ContactMe />);
    });

    it('should match Snapshot', () => {
        const wrapper = render(<ContactMe />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render an appropriate heading', () => {
        const { getByText } = render(<ContactMe />);
        expect(getByText('Contact')).toBeInTheDocument();
    });

    it('should have a link to the correct contact email', () => {
        const { getByText } = render(<ContactMe />);
        expect(getByText('mario.amaya.web@gmail.com')).toHaveAttribute('href', 'mailto:mario.amaya.web@gmail.com');
    });

    it('should render the correct phone number', () => {
        const { getByText } = render(<ContactMe />);
        expect(getByText('(619) 502-0073')).toBeInTheDocument();
    });

    it('should contain a link to the correct LinkedIn profile', () => {
        const { getByText } = render(<ContactMe />);
        expect(getByText('marioamayaweb')).toHaveAttribute('href', 'https://www.linkedin.com/in/marioamayaweb');
    });

    it('should contain a link to the correct GitHub profile', () => {
        const { getByText } = render(<ContactMe />);
        expect(getByText('MAmaya1')).toHaveAttribute('href', 'https://github.com/MAmaya1');
    });
});