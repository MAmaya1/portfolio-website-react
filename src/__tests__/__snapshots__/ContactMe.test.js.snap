// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe component should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxirZ cWwpYu"
      >
        <h1
          class="sc-AxiKw fXcBkv"
        >
          Contact 
          <span>
            Me
          </span>
        </h1>
        <h2
          class="sc-AxhCb fPoYls"
        >
          This is how you can reach me...
        </h2>
        <ul
          class="sc-AxhUy eKHqWi"
        >
          <li>
            <i
              class="far icon-bottom fa-envelope"
            />
             
            <a
              href="mailto:mario.amaya.web@gmail.com"
            >
              mario.amaya.web
              <wbr />
              @gmail.com
            </a>
          </li>
          <li>
            <i
              class="fas icon-bottom fa-phone"
            />
             (619) 502-0073
          </li>
          <li>
            <i
              class="fab icon-bottom fa-linkedin"
            />
             
            <a
              href="https://www.linkedin.com/in/marioamayaweb"
              rel="noopener noreferrer"
              target="_blank"
            >
              marioamayaweb
            </a>
          </li>
          <li>
            <i
              class="fab icon-bottom fa-github"
            />
             
            <a
              href="https://github.com/MAmaya1"
              rel="noopener noreferrer"
              target="_blank"
            >
              MAmaya1
            </a>
          </li>
        </ul>
      </div>
      <footer
        class="sc-AxjAm hnGFdg"
      >
        <h5>
          Mario Amaya | 
          <span>
            React Developer
          </span>
        </h5>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxirZ cWwpYu"
    >
      <h1
        class="sc-AxiKw fXcBkv"
      >
        Contact 
        <span>
          Me
        </span>
      </h1>
      <h2
        class="sc-AxhCb fPoYls"
      >
        This is how you can reach me...
      </h2>
      <ul
        class="sc-AxhUy eKHqWi"
      >
        <li>
          <i
            class="far icon-bottom fa-envelope"
          />
           
          <a
            href="mailto:mario.amaya.web@gmail.com"
          >
            mario.amaya.web
            <wbr />
            @gmail.com
          </a>
        </li>
        <li>
          <i
            class="fas icon-bottom fa-phone"
          />
           (619) 502-0073
        </li>
        <li>
          <i
            class="fab icon-bottom fa-linkedin"
          />
           
          <a
            href="https://www.linkedin.com/in/marioamayaweb"
            rel="noopener noreferrer"
            target="_blank"
          >
            marioamayaweb
          </a>
        </li>
        <li>
          <i
            class="fab icon-bottom fa-github"
          />
           
          <a
            href="https://github.com/MAmaya1"
            rel="noopener noreferrer"
            target="_blank"
          >
            MAmaya1
          </a>
        </li>
      </ul>
    </div>
    <footer
      class="sc-AxjAm hnGFdg"
    >
      <h5>
        Mario Amaya | 
        <span>
          React Developer
        </span>
      </h5>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
